service/replay.go

```diff
@@ -100,3 +100,44 @@ func (c *ReplayCache) Add(id string, salt []byte) bool {
 	c.active[hash] = empty{}
 	return !inArchive
 }
+
+// Resize adjusts the capacity of the ReplayCache.
+//
+// If the new capacity is less than the current capacity, and the number of
+// active handshakes exceeds the new capacity, then we move the excess to the
+// archive.
+func (c *ReplayCache) Resize(capacity int) {
+	if capacity > MaxCapacity {
+		panic("ReplayCache capacity would result in too many false positives")
```

Comment:
Please return an error and handle it instead of crashing the server. Config updates shouldn't crash the server (perhaps log error and keep the old config).